### PR TITLE
Core metrics functionality

### PR DESCRIPTION
**Pull request by znirzej**
Core metrics functionality + Unit tests

### CODE REVIEW

**Comment by znirzej** [comment]
@serathius @cezarygerard @kawych

File: custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		return nil, nil, err
+	}
+	stackdriverResponse, err := stackdriverRequest.Do()
+	if err != nil {
+		return nil, nil, err
+	}
+	return p.translator.GetCoreContainerMetricFromResponse(stackdriverResponse)
+}
+
+func (p *stackdriverCoreClient) getContainerRam(podsNames []string) (map[string]map[string]resource.Quantity, map[string]api.TimeInfo, error) {
```
Comment (serathius) [inline custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go:43]: RAM is an abbreviation, should be
```suggestion
func (p *stackdriverCoreClient) getContainerRAM(podsNames []string) (map[string]map[string]resource.Quantity, map[string]api.TimeInfo, error) {
```

File: custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+type stackdriverCoreClient struct {
+	translator *translator.Translator
+}
+
+func newClient(translator *translator.Translator) *stackdriverCoreClient {
+	return &stackdriverCoreClient{translator}
+}
+
+// TODO(holubowicz): handle a case when len(resourceNames) > oneOfMax
+func (p *stackdriverCoreClient) getContainerCpu(podsNames []string) (map[string]map[string]resource.Quantity, map[string]api.TimeInfo, error) {
```
Comment (serathius) [inline custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go:31]: CPU is an abbreviation, should be
```suggestion
func (p *stackdriverCoreClient) getContainerCPU(podsNames []string) (map[string]map[string]resource.Quantity, map[string]api.TimeInfo, error) {
```

File: custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		return nil, nil, err
+	}
+	stackdriverResponse, err := stackdriverRequest.Do()
+	if err != nil {
+		return nil, nil, err
+	}
+	return p.translator.GetCoreContainerMetricFromResponse(stackdriverResponse)
+}
+
+func (p *stackdriverCoreClient) getNodeCpu(nodesNames []string) (map[string]resource.Quantity, map[string]api.TimeInfo, error) {
```
Comment (serathius) [inline custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go:55]:
```suggestion
func (p *stackdriverCoreClient) getNodeCPU(nodesNames []string) (map[string]resource.Quantity, map[string]api.TimeInfo, error) {
```

File: custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	}
+	stackdriverResponse, err := stackdriverRequest.Do()
+	if err != nil {
+		return nil, nil, err
+	}
+	return p.translator.GetCoreContainerMetricFromResponse(stackdriverResponse)
+}
+
+func (p *stackdriverCoreClient) getContainerRam(podsNames []string) (map[string]map[string]resource.Quantity, map[string]api.TimeInfo, error) {
+	stackdriverRequest, err := p.translator.GetSDReqForContainersWithNames(podsNames, containerRamMetricName, ramMetricKind, ramNonEvictableLabel(), "")
```
Comment (serathius) [inline custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go:44]: Passing empty string looks wierd, you can either:
* add comment `"" /* namespace */` 
* create const `allNamespaces`

File: custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/coreprovider_test.go [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+	if err != nil {
+		t.Fatalf("Provider error: %s", err)
+	}
+
+	expectedMetrics := [][]metrics.ContainerMetrics{
+		{{"cont1", corev1.ResourceList{"cpu": contCpuRes["namespace1:pod1"]["cont1"], "memory": contRamRes["namespace1:pod1"]["cont1"]}}},
+	}
+	timeExpected := []api.TimeInfo{
+		{time1, time.Minute},
+	}
```
Comment (serathius) [inline custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/coreprovider_test.go:196]: Setup for those tests is too big and doesn't really show what's your realy testing

**Comment by serathius** [comment]
LGTM
/cc @kawych

**Review by kawych — COMMENTED** [review]
LGTM, Please address one nit I have and let me know to merge it.

**Pull request merged**